crates/chain/tests/common/tx_template.rs

```diff
@@ -0,0 +1,139 @@
+use std::collections::HashMap;
+
+use bdk_chain::{
+    indexed_tx_graph::IndexedTxGraph, keychain::KeychainTxOutIndex, ConfirmationHeightAnchor,
+};
+use bitcoin::{
+    hashes::Hash, locktime::absolute::LockTime, OutPoint, ScriptBuf, Sequence, Transaction, TxIn,
+    TxOut, Txid, Witness,
+};
+use miniscript::{Descriptor, DescriptorPublicKey};
+
+use super::TestKeychain;
+
+/// Transaction template.
+pub struct TxTemplate<'a, K, A> {
+    /// Uniquely identifies the transaction, before it can have a txid.
+    pub tx_name: &'a str,
+    pub inputs: &'a [TxInTemplate<'a>],
+    pub outputs: &'a [TxOutTemplate<K>],
+    pub anchors: &'a [A],
+    pub last_seen: Option<u64>,
+}
+
+impl<'a, K, A> Default for TxTemplate<'a, K, A> {
+    fn default() -> Self {
+        Self {
+            tx_name: Default::default(),
+            inputs: Default::default(),
+            outputs: Default::default(),
+            anchors: Default::default(),
+            last_seen: Default::default(),
+        }
+    }
+}
+
+impl<'a, K, A> Copy for TxTemplate<'a, K, A> {}
+
+impl<'a, K, A> Clone for TxTemplate<'a, K, A> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+#[allow(dead_code)]
+pub enum TxInTemplate<'a> {
+    /// This will give a random txid and vout.
+    Bogus,
+
+    /// This is used for coinbase transactions because they do not have previous outputs.
+    Coinbase,
+
+    /// Contains the `tx_name` and `vout` that we are spending. The rule is that we must only spend
+    /// a previous transaction.
+    PrevTx(&'a str, usize),
+}
+
+pub struct TxOutTemplate<K> {
+    pub value: u64,
+    pub owned_spk: Option<(K, u32)>, // some = derive a spk from K, none = random spk
+}
+
+#[allow(dead_code)]
+pub fn init_graph<'a>(
+    graph: IndexedTxGraph<ConfirmationHeightAnchor, KeychainTxOutIndex<TestKeychain>>,
+    descriptors: HashMap<TestKeychain, Descriptor<DescriptorPublicKey>>,
```

Comment:
I just realized we probably don't need to have multiple descriptors here. Since we are testing conflicts. At most, we just need to know whether the spk is tracked or not tracked (so a single descriptor is fine).

`KeychainTxOutIndex<()>` will do I think.

Reply:
Why do we need descriptors at all? You can just insert script pubkeys into `SpkTxOutIndex`. I am a little unsure as to why we need an indexed graph either. Doesn't transaction conflict handling exclusively in the domain of methods on `TxGraph` which are passed a `ChainOracle`.

Reply:
@LLFourn Good point.